pkg/sentry/kernel/semaphore/semaphore.go

```diff
@@ -171,10 +171,10 @@ func (r *Registry) FindOrCreate(ctx context.Context, key, nsems int32, mode linu
 	// Map semaphores and map indexes in a registry are of the same size,
 	// check map semaphores only here for the system limit.
 	if len(r.semaphores) >= setsMax {
-		return nil, syserror.EINVAL
+		return nil, syserror.ENOSPC
```

Comment:
Can you add a test for this?

Maybe somewhere around here?
https://github.com/google/gvisor/blob/master/test/syscalls/linux/semaphore.cc#L80-L102

Reply:
I added a test for number of semaphores per set, and number of system-wide semaphores. Number of system-wide semaphore sets seems untestable, as the system runs out of memory before exceeding the limit for sets.

Reply:
Ok, I think that may be good enough.